pkg/await/core_service.go

```diff
@@ -345,24 +345,74 @@ func (sia *serviceInitAwaiter) errorMessages() []string {
 	return messages
 }
 
+// isHeadlessService checks if the Service has a defined .spec.clusterIP
+func (sia *serviceInitAwaiter) isHeadlessService() bool {
+	clusterIP, _ := openapi.Pluck(sia.service.Object, "spec", "clusterIP")
+	return clusterIP == v1.ClusterIPNone
+}
+
+// isExternalNameService checks if the Service type is "ExternalName"
+func (sia *serviceInitAwaiter) isExternalNameService() bool {
```

Comment:
Not a huge deal, but we might consider making these just normal functions instead of methods. I personally find it harder to reason about structs with lots of methods.

Reply:
Agreed, but I expect to refactor this soon anyway as part of #343